command/deployment_status.go

```diff
@@ -358,3 +391,247 @@ func formatDeploymentGroups(d *api.Deployment, uuidLength int) string {
+// but only used for tty and non-Windows machines since glint doesn't work with
+// cmd/PowerShell and non-interactive interfaces
+// Margins are used to match the text alignment from job run
+func (c *DeploymentStatusCommand) ttyMonitor(client *api.Client, deployID string, index uint64, verbose bool) {
+	var length int
+	if verbose {
+		length = fullId
+	} else {
+		length = shortId
+	}
+
+	d := glint.New()
+	spinner := glint.Layout(
+		components.Spinner(),
+		glint.Text(fmt.Sprintf(" Deployment %q in progress...", limit(deployID, length))),
+	).Row().MarginLeft(2)
+	refreshRate := 100 * time.Millisecond
+
+	d.SetRefreshRate(refreshRate)
+	d.Set(spinner)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go d.Render(ctx)
+	defer cancel()
+
+	q := api.QueryOptions{
+		AllowStale: true,
+		WaitIndex:  index,
+		WaitTime:   2 * time.Second,
+	}
+
+	var statusComponent *glint.LayoutComponent
+	var endSpinner *glint.LayoutComponent
+
+UPDATE:
+	for {
+		deploy, meta, err := client.Deployments().Info(deployID, &q)
+		if err != nil {
+			d.Append(glint.Style(
+				glint.Text(fmt.Sprintf("%s: Error fetching deployment", formatTime(time.Now()))),
+				glint.Color("red"),
+			))
+			d.RenderFrame()
+			return
+		}
+
+		status := deploy.Status
+		statusComponent = glint.Layout(
+			glint.Text(""),
+			glint.Text(formatTime(time.Now())),
+			// Use colorize to render bold text in formatDeployment function
+			glint.Text(c.Colorize().Color(formatDeployment(client, deploy, length))),
+		)
+
+		if verbose {
+			allocComponent := glint.Layout(glint.Style(
+				glint.Text("Allocations"),
+				glint.Bold(),
+			))
+
+			allocs, _, err := client.Deployments().Allocations(deployID, nil)
+			if err != nil {
+				allocComponent = glint.Layout(
+					allocComponent,
+					glint.Style(
+						glint.Text("Error fetching allocations"),
+						glint.Color("red"),
+					),
+				)
+			} else {
+				allocComponent = glint.Layout(
+					allocComponent,
+					glint.Text(formatAllocListStubs(allocs, verbose, length)),
+				)
+			}
+
+			statusComponent = glint.Layout(
+				statusComponent,
+				glint.Text(""),
+				allocComponent,
+			)
+		}
+
+		statusComponent = glint.Layout(statusComponent).MarginLeft(4)
+		d.Set(spinner, statusComponent)
+
+		endSpinner = glint.Layout(
+			components.Spinner(),
+			glint.Text(fmt.Sprintf(" Deployment %q %s", limit(deployID, length), status)),
+		).Row().MarginLeft(2)
+
+		switch status {
+		case structs.DeploymentStatusFailed:
+			if hasAutoRevert(deploy) {
+				// Separate rollback monitoring from failed deployment
+				d.Set(
+					endSpinner,
+					statusComponent,
+					glint.Layout(glint.Text("")),
+				)
+
+				// Wait for rollback to launch
+				time.Sleep(1 * time.Second)
+				rollback, _, err := client.Jobs().LatestDeployment(deploy.JobID, nil)
+
+				if err != nil {
+					d.Append(glint.Style(
+						glint.Text(fmt.Sprintf("%s: Error fetching deployment of previous job version", formatTime(time.Now()))),
```

Comment:
Maybe we should use language to make it clear this is a revert?

```suggestion
						glint.Text(fmt.Sprintf("%s: Error fetching rollback deployment", formatTime(time.Now()))),
```